src/stores/FiltersHistoryStore.ts

```diff
@@ -113,43 +115,33 @@ class FiltersHistoryStore {
 	}
 
 	@action
-	public onEventFilterSubmit = async (newFilters: FiltersHistoryType<EventFilterState>) => {
-		if (isEmptyFilter(newFilters.filters)) return;
-
-		const { type, timestamp } = newFilters;
-		const equilizedFilter = getNonEmptyFilters(newFilters.filters);
-
-		if (this.eventsHistory.some(({ filters }) => isEqual(filters, equilizedFilter))) return;
+	public onEventFilterSubmit = async (newItem: FiltersHistoryType<EventFilterState>) => {
+		if (isEmptyFilter(newItem.filters)) return;
 
-		const filter: FiltersHistoryType<EventFilterState> = {
-			timestamp,
-			type,
-			filters: equilizedFilter,
-			isPinned: false,
-		};
+		const filter = getEquilizedItem(newItem);
 
-		this.indexedDb.addDbStoreItem(IndexedDbStores.FILTERS_HISTORY, filter);
-		this.filterHistory = [...this.filterHistory, filter];
+		if (this.initialized) {
+			this.addEventHistoryItem(filter);
+		} else {
+			this.init().then(() => {
+				this.addEventHistoryItem(filter);
+			});
+		}
 	};
 
 	@action
-	public onMessageFilterSubmit = async (newFilters: FiltersHistoryType<MessageFilterState>) => {
-		if (isEmptyFilter(newFilters.filters)) return;
-
-		const { type, timestamp } = newFilters;
-		const equilizedFilter = getNonEmptyFilters(newFilters.filters);
+	public onMessageFilterSubmit = async (newItem: FiltersHistoryType<MessageFilterState>) => {
+		if (isEmptyFilter(newItem.filters)) return;
 
-		if (this.messagesHistory.some(({ filters }) => isEqual(filters, equilizedFilter))) return;
+		const filter = getEquilizedItem(newItem);
 
-		const filter: FiltersHistoryType<MessageFilterState> = {
-			timestamp,
-			type,
-			filters: equilizedFilter,
-			isPinned: false,
-		};
-
-		this.indexedDb.addDbStoreItem(IndexedDbStores.FILTERS_HISTORY, filter);
-		this.filterHistory = [...this.filterHistory, filter];
+		if (this.initialized) {
+			this.addMessageHistoryItem(filter);
+		} else {
+			this.init().then(() => {
```

Comment:
we call this function twice now.
let's make initialized observable property
then we can use `when` from mobx
`await when(() => this.initialized)`
then add